### ISSUE TITLE
  Cannot start docker service sockjs , sockjs , api ,  scheduler，hubshell, poller

### ISSUE THREAD

**Issue by zhenfdfs** [mistio/mist-ce#904]
ERROR: for sockjs  Cannot start service sockjs: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file
or directory"

ERROR: for celery  Cannot start service celery: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file
or directory"

ERROR: for api  Cannot start service api: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file or dir
ectory"

ERROR: for scheduler  Cannot start service scheduler: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such
 file or directory"

ERROR: for hubshell  Cannot start service hubshell: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such f
ile or directory"

ERROR: for poller  Cannot start service poller: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file
or directory"
ERROR: Encountered errors while bringing up the project.

**Comment by zhenfdfs** [comment]
docker info

Containers: 20
 Running: 10
 Paused: 0
 Stopped: 10
Images: 15
Server Version: 17.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.27-moby
Operating System: Alpine Linux v3.5
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 1.934 GiB
Name: moby
ID: QQNZ:OYYW:TR77:SHWL:BEWZ:EJTT:NWAB:XVK4:VQTO:KVJU:KKLH:XGC5
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 133
 Goroutines: 82
 System Time: 2017-06-14T01:27:23.194218Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Registry Mirrors:
 http://ebd9bc0e.m.daocloud.io
Live Restore Enabled: false

**Comment by zhenfdfs** [comment]
my environment as follow
Centos 7.2  with kernel 3.10.0-514.21.1.el7.x86_64
Docker version 17.05.0-ce, build 89658be
docker-compose version 1.14.0-rc2, build 24dae73


this error happed when i run docker-compose up -d.  but it can not reproduce when i run these docker with command " docker run --name celery --net="mistio_default" -d mist/mist:io-staging bin/wait-all bin/celery"

**Comment by mgogoulos** [comment]
Hi @zhenfdfs , 

what version are you trying? Have you used one from https://github.com/mistio/mist.io/releases ?

Cheers

**Comment by dalareo** [comment]
I have the same error using 2.0 release:

```
docker info
Containers: 22
 Running: 11
 Paused: 0
 Stopped: 11
Images: 32
Server Version: 17.05.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 212
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9048e5e50717ea4497b757314bad98ea3763c145
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.10.0-25-lowlatency
Operating System: Ubuntu 17.04
OSType: linux
Architecture: x86_64
CPUs: 6
Total Memory: 7.531GiB
Name: xxxxxxxxxx
ID: 5ETT:TBGA:OG4T:BJ3J:MO3V:FTWH:MPEM:M6XS:LSSG:XSKJ:CORW:CTH6
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: xxxxxx
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Comment by dimrozakis** [comment]
@zhenfdfs I see that you closed this issue. Can you please let us know how you resolved it?

@dalareo can you please provide some more information:
1. OS/Distro name and version
2. Docker compose version (`docker-compose --version`)

**Comment by dimrozakis** [comment]
@dalareo In an attempt to reproduce your issue, I spun up a digital ocean VM running `Ubuntu 17.04`.

```bash
> uname -a
Linux dr-debug 4.10.0-24-generic #28-Ubuntu SMP Wed Jun 14 08:14:34 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```

I installed `docker` with:
```bash
> curl -sSL https://get.docker.com/ | sh
```

Docker info is:
```
> docker info
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 17.05.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 0
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9048e5e50717ea4497b757314bad98ea3763c145
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.10.0-24-generic
Operating System: Ubuntu 17.04
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.794GiB
Name: dr-debug
ID: XXW6:WKWX:NGEP:6MUE:ZF4A:R66D:CJ4L:RQNG:Y5P6:YM5U:S5IH:74NU
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

Then I installed `docker-compose` with:
```bash
> apt install -y python-pip
> pip install docker-compose
```

Docker compose version is:
```bash
> docker-compose --version
docker-compose version 1.14.0, build c7bdf9e
```

Then I did a copy-paste of the instruction provided on the `v2.0.0` [release page](https://github.com/mistio/mist.io/releases/tag/v2.0.0).
```bash
> mkdir mist.io && cd mist.io
> wget https://github.com/mistio/mist.io/releases/download/v2.0.0/docker-compose.yml
> docker-compose up -d
```

It took some time to pull all images, but after it completed, everything seems to be working fine:
```bash
> docker-compose ps
            Name                           Command               State                                Ports                             
---------------------------------------------------------------------------------------------------------------------------------------
mistio_api_1                    /mist.api/bin/docker-init  ...   Up       80/tcp                                                        
mistio_beat_1                   /mist.api/bin/docker-init  ...   Up                                                                     
mistio_celery_1                 /mist.api/bin/docker-init  ...   Up                                                                     
mistio_elasticsearch-manage_1   ./scripts/add_templates.py       Exit 0                                                                 
mistio_elasticsearch_1          /docker-entrypoint.sh elas ...   Up       9200/tcp, 9300/tcp                                            
mistio_hubshell_1               /mist.api/bin/docker-init  ...   Up                                                                     
mistio_kibana_1                 /docker-entrypoint.sh kibana     Up       5601/tcp                                                      
mistio_landing_1                sh /entry.sh                     Up       80/tcp                                                        
mistio_logstash_1               /docker-entrypoint.sh logs ...   Up                                                                     
mistio_mailmock_1               mailmock -o 8025 -o /var/mail    Up       8025/tcp                                                      
mistio_memcached_1              docker-entrypoint.sh memcached   Up       11211/tcp                                                     
mistio_mongodb_1                docker-entrypoint.sh mongod      Up       27017/tcp                                                     
mistio_nginx_1                  nginx -g daemon off;             Up       0.0.0.0:80->80/tcp                                            
mistio_poller_1                 /mist.api/bin/docker-init  ...   Up                                                                     
mistio_rabbitmq_1               docker-entrypoint.sh rabbi ...   Up       15671/tcp, 15672/tcp, 25672/tcp, 4369/tcp, 5671/tcp, 5672/tcp 
mistio_scheduler_1              /mist.api/bin/docker-init  ...   Up                                                                     
mistio_socat_1                  socat TCP4-LISTEN:2375,for ...   Up       2375/tcp                                                      
mistio_sockjs_1                 /mist.api/bin/docker-init  ...   Up       8081/tcp                                                      
mistio_ui_1                     sh /entry.sh                     Up       80/tcp                                                        
```

So I can't reproduce the issue you've been facing. Please provide as much information as possible to track this down.

**Comment by dalareo** [comment]
Hi @dimrozakis I am using Ubuntu 17.04 with docker-compose version 1.12.0, build b31ff33

**Comment by dalareo** [comment]
I have upgraded to docker-compose 1.14 but get:

```
Creating mistio_api_1 ... error

ERROR: for mistio_api_1  Cannot start service api: oci runtime error: container_linux.go:247: starting container process caused "exeCreating mistio_beat_1 ... error

ERROR: for mistio_beat_1  Cannot start service beat: oci runtime error: container_linux.go:247: starting container process caused "eCreating mistio_celery_1 ... error

ERROR: for mistio_celery_1  Cannot start service celery: oci runtime error: container_linux.go:247: starting container process causeCreating mistio_scheduler_1 ... error

ERROR: for mistio_scheduler_1  Cannot start service scheduler: oci runtime error: container_linux.go:247: starting container processCreating mistio_poller_1 ... error

ERROR: for mistio_poller_1  Cannot start service poller: oci runtime error: container_linux.go:247: starting container process causeCreating mistio_sockjs_1 ... error

ERROR: for mistio_sockjs_1  Cannot start service sockjs: oci runtime error: container_linux.go:247: starting container process causeCreating mistio_hubshell_1 ... error

ERROR: for mistio_hubshell_1  Cannot start service hubshell: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file or directory"

ERROR: for sockjs  Cannot start service sockjs: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file or directory"

ERROR: for beat  Cannot start service beat: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file or directory"

ERROR: for celery  Cannot start service celery: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file or directory"

ERROR: for api  Cannot start service api: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file or directory"

ERROR: for scheduler  Cannot start service scheduler: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file or directory"

ERROR: for hubshell  Cannot start service hubshell: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file or directory"

ERROR: for poller  Cannot start service poller: oci runtime error: container_linux.go:247: starting container process caused "exec: \"/mist.api/bin/docker-init\": stat /mist.api/bin/docker-init: no such file or directory"
ERROR: Encountered errors while bringing up the project.
```

**Comment by dimrozakis** [comment]
This is weird.

The container images have an init script in `/mist.api/bin/docker-init` which runs before anything else in the containers. The error message you pasted suggests that no such script can be found inside the containers, which doesn't make much sense.

Can you please run `docker run mist/mist:io-v2-0-0 ls -l /mist.api/bin/docker-init`? It should return
```
-rwxr-xr-x    1 root     root           705 Jun 20 15:02 /mist.api/bin/docker-init
```

Can you also copy/paste your `docker-compose.yml` version here?

**Comment by dalareo** [comment]
Yes, it returns that line.

My docker-compose.yml

```
# Docker compose definition to run mist.io in production mode, so no mounted
# code, no dev containers etc. Only this single file is required.

version: '2.0'

services:


  mongodb:
    image: mongo:3.2
    restart: on-failure:5
    volumes:
      - mongodb:/data/db:rw

  rabbitmq:
    image: rabbitmq:3.6.6-management
    restart: on-failure:5

  memcached:
    image: memcached
    restart: on-failure:5


  elasticsearch:
    image: elasticsearch:5.1.2
    restart: on-failure:5
    volumes:
      - elasticsearch:/usr/share/elasticsearch/data:rw

  logstash:
    image: mist/logstash:v2-0-0
    restart: on-failure:5
    depends_on:
      - elasticsearch
      - rabbitmq

  elasticsearch-manage:
    image: mist/elasticsearch-manage:v2-0-0
    command: ./scripts/add_templates.py
    restart: on-failure:5
    depends_on:
      - elasticsearch
      - rabbitmq

  kibana:
    image: kibana:5.1.2
    environment:
      ELASTICSEARCH_URL: http://elasticsearch:9200
    restart: on-failure:5
    depends_on:
      - elasticsearch


  socat:
    image: mist/docker-socat
    restart: on-failure:5
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:rw

  mailmock:
    image: mist/mailmock
    restart: on-failure:5


  celery: &backend
    image: mist/mist:io-v2-0-0
    depends_on:
      - memcached
      - mongodb
      - elasticsearch
      - rabbitmq
    volumes:
      - ./config:/etc/mist.io
    environment:
      SETTINGS_FILE: /etc/mist.io/settings.py
    stdin_open: true
    tty: true
    restart: on-failure:5
    command: bin/wait-all bin/celery

  scheduler:
    <<: *backend
    command: bin/wait-all bin/scheduler

  poller:
    <<: *backend
    command: bin/wait-all bin/poller

  beat:
    <<: *backend
    command: bin/wait-all bin/beat

  hubshell:
    <<: *backend
    command: bin/wait-all bin/hubshell

  api:
    <<: *backend
    command: bin/wait-all bin/uwsgi
    expose:
      - 80

  sockjs:
    <<: *backend
    command: bin/wait-all bin/sockjs
    expose:
      - 8081


  ui:
    image: mist/ui:io-v2-0-0
    command: sh /entry.sh
    stdin_open: true
    tty: true
    restart: on-failure:5
    expose:
      - 80

  landing:
    image: mist/landing:io-v2-0-0
    command: sh /entry.sh
    stdin_open: true
    tty: true
    restart: on-failure:5
    expose:
      - 80


  nginx:
    image: mist/nginx:v2-0-0
    restart: on-failure:5
    ports:
      - 80:80
    depends_on:
      - api
      - sockjs
      - landing
      - ui

volumes:
  elasticsearch: {}
  mongodb: {}
```

**Comment by dimrozakis** [comment]
I just don't get it.

As you can see, `celery`, one of the containers that fails because it can't find `/mist.api/bin/docker-init` uses the same `mist/mist:io-v2-0-0` image where we just manually verified that the script is present.

1. I assume you installed v2 using the instruction in github's releases page, not by cloning the repo, right?
2. Can you paste the result of `ls -l` from inside the directory where your docker-compose lies?
3. Is the machine where you're experiencing the issue a cloud VM? If so can you provide more info the the cloud provider and image so that I can attempt to reproduce the issue?

**Comment by dalareo** [comment]
Yes, I downloaded v2 with the script of the repo in an empty folder. The result of ls -l is:

```
drwxr-xr-x 2 root    root    4096 Jun 26 18:28 config
-rw-r--r-- 1 usuario usuario 2524 Jun 21 10:28 docker-compose.yml
```

I am using a PC. It is a phisical machine.

**Comment by dimrozakis** [comment]
A quick google search returned https://github.com/docker/compose/issues/4039 which suggests trying to run `docker-compose up -d` using `sudo`. Can you please try that and report if it worked?

**Comment by dimrozakis** [comment]
Also, if that fails, can you try restarting the docker daemon and/or rebooting your computer to see if perhaps that fixes it? I know this sounds like the classical "have you tried turning it off and on again" suggestion, but I'm running out of ideas.

**Comment by mgogoulos** [comment]
I have also run the 2.0.0 release (https://github.com/mistio/mist.io/releases/tag/v2.0.0) on Ubuntu 17.04 with docker and docker-compose installed out of the official ubuntu repository - apt-get install docker-compose -y . This has installed docker 1.12.6 and docker-compose version 1.8.0. 

I then tried to start docker-compose as user and also as root, and do actions as stop/restart again both as user and as root. In all cases I haven't been able to reproduce the issue, and docker-compose can start/stop/restart containers all times. Only thing I made sure that user belongs to the docker group, by editing /etc/groups.

My only thought at this point is that there's an error with permissions when docker-compose is run as user, so I suggest you either run docker-compose as root, or as @dimrozakis suggest as a sudo user, with the sudo command ahead.
